### PR TITLE
Adjoh/adal update common

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
Please make sure every Pull Request has the following information:

* A reference to a related issue in your repository (mandatory)
* A description of the changes proposed in the pull request
* @mentions of the person required for reviewing proposed changes (optional)

### CODE REVIEW

File: userappwithbroker/userapp-rules.pro [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Add project specific ProGuard rules here.
```
Comment (rpdome) [inline userappwithbroker/userapp-rules.pro:1]: Do we need this? (given that user app is our internal test app... not having this means one less thing to maintain.)

File: adal/adal-rules.pro [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+-dontskipnonpubliclibraryclasses
+
+##---------------Begin: proguard configuration for Common  --------
+-keep,includedescriptorclasses class com.microsoft.identity.common.exception.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.broker.** { *; }
```
Comment (rpdome) [inline adal/adal-rules.pro:23]: I'm thinking if we could just do 

-keep,includedescriptorclasses class com.microsoft.identity.common.**

This way, we don't have to maintain this file (and the one in MSAL) much when we introduce new stuffs in common.

File: adal/adal-rules.pro [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.cache.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.logging.internal.dto.** { *; }
```
Comment (rpdome) [inline adal/adal-rules.pro:25]: missing .common

File: adal/adal-rules.pro [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+-keep,includedescriptorclasses class com.microsoft.identity.internal.logging.** { *; }
```
Comment (rpdome) [inline adal/adal-rules.pro:26]: missing .common

**Review by rpdome — APPROVED** [review]
left a few comments.

File: userappwithbroker/userapp-rules.pro (hunk 1 above)
Comment (AdamBJohnsonx) [reply userappwithbroker/userapp-rules.pro:1]: I think we should dog food the ProGuard rules.

File: adal/adal-rules.pro (hunk 2 above)
Comment (AdamBJohnsonx) [reply adal/adal-rules.pro:23]: The idea here is that we don't use most of common in ADAL, so we should drop as much of it as we can.

**Pull request closed without merging**